### PR TITLE
fix: aggkit-proxy: bali integration: bridge tracker activity endpoint, GER settlement/injection fixes

### LINKED ISSUE #1818: bug: proxy: step WaitingGERInjection as result have the L1 block
# Bug Report

The block of `WaitingGERInjection`  is the same as `WaitingGERInjection`. So can't be used to calculte timing  and doesn't give any new data. 
To get this info is calling to bridge service `/bridge/v1/injected-l1-info-leaf` this must provide more information

```

{
      "step_index": 1,
      "step_name": "WaitingGERInjection",
      "status": "done",
      "start_date": "2026-08-31T17:43:20.307055948+02:00",
      "end_date": "2026-08-31T17:44:05.091708378+02:00",
      "result": {
        "ger": "0x330d1f1546dc784aa465fdf83fb9d88e0a3778064d74e182c1dfb803ef155c16",
        "block_number": 11606405,
        "block_timestamp": 1788188124
      }
    },

(...)

{
      "step_index": 1,
      "step_name": "WaitingGERInjection",
      "status": "done",
      "start_date": "2026-08-31T17:43:20.307055948+02:00",
      "end_date": "2026-08-31T17:44:05.091708378+02:00",
      "result": {
        "ger": "0x330d1f1546dc784aa465fdf83fb9d88e0a3778064d74e182c1dfb803ef155c16",
        "block_number": 11606405,
        "block_timestamp": 1788188124
      }
    },
```
# Proposed fix
- Add new fields to bridge_service end-point `injected-l1-info-leaf`: 
     - injected_l2_block_num
     - injected_l2_block_timestamp
These fields are optional (to keep compatibility) 

- Modify result for WaitingGERInjection
```
      "result": {
        "L1InfoTreeLeaf": {
        "ger": "0x330d1f1546dc784aa465fdf83fb9d88e0a3778064d74e182c1dfb803ef155c16",
        "block_number": 11606405,
        "block_timestamp": 1788188124
       }, 
      "L2InjectedGER": {
        "block_number": xxx,
        "block_timestamp": yyy
      }
      }
```
L2InjectedGER is optional

### PR DESCRIPTION
**Pull request by joanestebanr**
## 🔄 Changes Summary
- **`feat(bridgetracker)`**: new `GET /tracker/v1/activity/from/{from_address}` endpoint that scans every bridge service known to the `bridgeservicefinder` (via new `Finder.NetworkIDs()`) for bridges sent by an address and resolves each one's claim state (`ActivityCache`, `ActivitySource`, `ActivityQuerier`/`ActivityBridgeScanner`/`ActivityClaimChecker` ports). Opt-in via `includeTracking=true` to also register still-unclaimed bridges with the tracker. Endpoint is only registered when wired (`proxy/cmd/run.go`), and swagger docs are regenerated.
- **`feat(bridgetracker)`**: `StepClaimed` is now its own tracked step with a dedicated `ClaimedResolver` that fetches the claim tx/block from the destination network's bridge service, decoupled from `StepWaitingClaim`, which now checks `isClaimed()` on-chain directly through a new `ClaimChecker` port instead of waiting on the indexed claim record — faster and authoritative, at the cost of not carrying claim tx details itself (that's what the new `Claimed` step result is for). The on-chain binding/cache logic is factored out of `ActivitySource` into `sources/claim_checker.go` (`contractClaimCheckers`) and shared by both the tracker engine and the activity endpoint.
- **`fix(bridgetracker)`**: introduce `domain.ScannedBridge` to track which network's bridge service actually reported a scanned bridge (`NetworkID`), distinct from `Bridge.OriginNetwork` (the bridged *asset*'s origin network) — the two diverge when an asset is re-bridged across more than one hop, which was feeding the wrong `sourceBridgeNetwork` into `isClaimed()` and the wrong network into `TrackingID`/the activity endpoint's `bridge_network_id` for such bridges.
- **`feat(bridgetracker)`**: new `GET /bridge-address[/{network_id}]` endpoint, resolving the bridge contract address for one network or every network currently known — opt-in via `Config.BridgeAddressResolver`, wired in `proxy/cmd/run.go` off `bridgeservicefinder.Finder` directly. Backed by a new `Config.BridgeAddress` override map on `bridgeservicefinder.Finder` itself (priority: per-network entry, then `[0]` as the default, then the rollup manager's own on-chain `BridgeAddress()`).
- **`feat(bridgetracker)`**: `ClaimResult` now also carries `BlockTimestamp` alongside `BlockNumber`. `InjectedGERResult` is restructured into `L1InfoTreeLeaf` (the L1 `UpdateL1InfoTree`/`UpdateL1InfoTreeV2` event that produced the covering leaf) and an optional `L2InjectedGER` (the actual L2 block/timestamp the GER was injected at on the destination network) — fixing #1818, where the L1 block was returned in the L2 field's place. When the destination's bridge-service instance doesn't report the L2 injection block itself (`injected_l2_block_num`/`injected_l2_block_timestamp` on `GET /bridge/v1/injected-l1-info-leaf`), `GERSource` falls back to scanning that network's own `GlobalExitRootManagerL2` contract for the `UpdateHashChainValue` event backwards in chunks, via the new `Tracker.L2GlobalExitRootAddress` per-network contract address map (a workaround-only fallback, see its doc) — bounded by a new `Tracker.L2InjectionLookbackBlocks` (default 1,000 blocks) instead of always walking back to genesis.
- **`fix(bridgetracker)`**: resolve the settled GER correctly when the settlement tx has no `UpdateL1InfoTree` event: walk backwards on L1 for the most recent earlier one instead of surfacing "not ready" forever, excluding same-block logs from a later transaction, and querying `eth_getLogs` in chunks that respect providers' 10,000-block range cap.
- **`fix(bridgetracker)`**: gate certificate settlement on the settlement tx actually being visible on L1 (not just the agglayer client reporting `Settled`), avoiding a premature `TransactionReceipt` lookup that would otherwise fail.

## ⚠️ Breaking Changes
- None. `Config.BridgeAddressResolver` and the new `bridgetracker` activity/bridge-address ports are additive/opt-in.

## 📋 Config Updates
- 🧾 New optional `BridgeServiceFinder.BridgeAddress` map (networkID → address override), default empty — consulted by the new `GET /bridge-address[/{network_id}]` endpoint before falling back to the rollup manager's on-chain `BridgeAddress()`. That endpoint itself is gated by `Config.BridgeAddressResolver` (Go-level wiring, not a TOML key); unset leaves both routes unregistered.
- 🧾 New optional `Tracker.ActivityIdleTimeout` (defaults to `IdleTimeout` if unset) — idle timeout for the activity endpoint's own cache.
- 🧾 New optional `Tracker.L2GlobalExitRootAddress` (networkID → `GlobalExitRootManagerL2` address map, default empty) and `Tracker.L2InjectionLookbackBlocks` (default `1000`) — workaround-only fallback for a destination network whose bridge-service instance predates L2 injection block reporting (#1819); defaults leave behavior unchanged for everyone else.

  ```toml
  [BridgeServiceFinder.BridgeAddress]
  # optional per-network override, consulted in this priority order by Finder.BridgeAddress:
  # BridgeAddress[networkID], then BridgeAddress[0] (default for every other network), then the
  # rollup manager's own on-chain BridgeAddress()
  0 = "0xShared_L1_bridge_address_used_as_the_default"
  5 = "0xNetwork5s_own_bridge_address"

  [Tracker]
  ActivityIdleTimeout = "30m"
  L2InjectionLookbackBlocks = 1000

  # Workaround only: uncomment for a destination network whose bridge-service instance does not
  # report the L2 block a covering GER was injected at.
  # [Tracker.L2GlobalExitRootAddress]
  # 1 = "0x..."
  ```

## ✅ Testing
- 🤖 **Automatic**: `go build ./...` and `go test ./bridgetracker/... ./bridgeservicefinder/... ./autoclaim/... ./proxy/...` pass, including new/updated regression tests for the activity endpoint (`activity_test.go`, `sources/activity_test.go`, `cache_test.go`), the claimed-step split (`resolve_steps_test.go`, `engine_test.go`), the new bridge-address endpoint (`bridge_address_test.go`), the settlement GER backwards-search fix (`settlement_test.go`), and the L2 injection fallback/lookback cap (`sources_test.go`'s `TestGERSourceInjectedGER_FallsBackToL2Scan` and `TestFindL2InjectionBlockBackwards`, `proxy/config/config_test.go`).

## 🐞 Issues
- Fixes #1818.

## 🔗 Related PRs
- None.

## 📝 Notes
- This branch's history still contains commits for `bridgeservicefinder`'s `IgnoreNetworkIDs` and the `l2gersync` `eth_getLogs` chunk-cap fix (originally developed alongside this work) — both landed directly on `develop` in the meantime (#1810, #1813), so neither shows up in this PR's diff anymore despite still being present in the branch's git history.

### CODE REVIEW

**Review by arnaubennassar — DISMISSED** [review]

**Comment by joanestebanr** [comment]
@claude review

File: bridgetracker/sources/activity.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		svc, err := s.services.aggkitBridgeClientFor(networkID)
+		if err != nil {
+			return nil, fmt.Errorf("resolving bridge service client for network %d: %w", networkID, err)
+		}
+
+		items, err := fetchAllBridgesFrom(ctx, svc, networkID, addr, activityPageSize)
+		if err != nil {
+			return nil, fmt.Errorf("fetching bridges from %s on network %d: %w", fromAddress, networkID, err)
+		}
+		all = append(all, items...)
```
[the remark this answers is not in the document]
Comment (joanestebanr) [reply bridgetracker/sources/activity.go]: fixed commit 7c1afb1f

File: bridgetracker/sources/settlement.go [hunk 2]
```diff
@@ -110,8 +137,82 @@ func (s *SettlementSource) SettlementGERUpdate(
...
+// domain.ErrBadSettlementTx (Permanent — see the sentinel's own doc)
+func (s *SettlementSource) findEventUpdateL1InfoTreeBackwards(
+	ctx context.Context, client aggkittypes.BaseEthereumClienter, fromBlock uint64,
+) (*updateL1InfoTreeEvent, error) {
+	toBlock := fromBlock
+	for {
+		fromBlockChunk := uint64(0)
+		if toBlock > l1InfoTreeBackwardsSearchChunkSize {
+			fromBlockChunk = toBlock - l1InfoTreeBackwardsSearchChunkSize
+		}
```
[the remark this answers is not in the document]
Comment (joanestebanr) [reply bridgetracker/sources/settlement.go]: Fixed in 0c021aab: added the missing `+1` (and widened the guard to `toBlock >= chunkSize`) so each backwards chunk is exactly `l1InfoTreeBackwardsSearchChunkSize` blocks instead of chunkSize+1.

**Comment by joanestebanr** [comment]
Fixed the off-by-one in `findEventUpdateL1InfoTreeBackwards` (Issue 1 in the review above, same as Codex's discussion_r3872865287) in 0c021aab — each backwards chunk now spans exactly `l1InfoTreeBackwardsSearchChunkSize` blocks (`toBlock - chunkSize + 1`), instead of chunkSize+1.

File: bridgetracker/sources/settlement.go [hunk 3]
```diff
@@ -110,8 +137,82 @@ func (s *SettlementSource) SettlementGERUpdate(
...
+		})
+		if err != nil {
+			return nil, fmt.Errorf("fetching UpdateL1InfoTree logs from block %d to %d: %w",
+				fromBlockChunk, toBlock, err)
+		}
+
+		if len(logs) > 0 {
+			// FilterLogs returns logs in ascending block/log-index order, so the last one is the
+			// most recent event in this chunk — the closest one at or before fromBlock
+			last := logs[len(logs)-1]
```
[the remark this answers is not in the document]
Comment (joanestebanr) [reply bridgetracker/sources/settlement.go:230]: Fixed in ce78daba: threaded `settlementLogIndex` through to `findEventUpdateL1InfoTreeBackwards` and filter out any log sharing `fromBlock` with the settlement at or after that index before picking the latest one — mirroring the position filtering `GERSource.FindFirstL1InfoTreeAfterBlock` already does in ger.go, just backwards. Added a dedicated test (`TestFindEventUpdateL1InfoTreeBackwardsExcludesLaterLogInSettlementBlock`) pinning it.

**Comment by joanestebanr** [comment]
Fixed the same-block GER exclusion issue (also flagged by Codex as discussion_r3872865292) in ce78daba — `findEventUpdateL1InfoTreeBackwards` now filters out any UpdateL1InfoTree log sharing the settlement's block at or after its own log index before picking the latest one.

**Comment by joanestebanr** [comment]
Addressed item 6 (edge case test coverage for the claim step split) in a6036601: added `TestResolveStepsClaimedNotIndexedYet` in domain/resolve_steps_test.go, and extended `TestEngineLifecycleL2ToL2` in engine_test.go to tick through the window where isClaimed() goes true but the bridge service hasn't indexed the claim tx yet — pinning that StepWaitingClaim completes (Done) while StepClaimed stays its own current step (InProgress, no result) until ClaimFor resolves on a later tick.

**Review by arnaubennassar — APPROVED** [review]
